### PR TITLE
Add support for Raw files previews

### PR DESCRIPTION
**Pull request by oparoz**
#### Solution for

https://github.com/owncloud/core/issues/13650
https://github.com/owncloud/apps/issues/439
https://github.com/owncloud/apps/issues/1113
#### Supported formats:
- Sony Alpha RAW .arw
- Canon RAW 2 .cr2
- Digital Camera Raw .dcr
- Digital Negative .dng
- EPSON RAW .erf
- Intelligent Image Quality RAW .iiq
- Kodak DC40/DC50 RAW .kdc
- Kodak DC25 RAW .k25
- Nikon RAW .nef
- Mamiya RAW .mef
- Olympus RAW .orf
- Panasonic RAW .rw2
- PENTAX RAW .pef
- Fujifilm RAW .raf
- Sony RAW File .srf .sr2
- SIGMA RAW .xrf
#### Requirements
- ImageMagick
- ufraw as an ImageMagick delegate for DNG
- ufraw-batch must be reachable from PHP
- PR https://github.com/owncloud/core/pull/13654, which is a fix for https://github.com/owncloud/core/issues/12282
- A gallery app which supports all media types supported by ownCloud

### CODE REVIEW

File: lib/private/preview/raw.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+// Supports many file extensions linked to RAW
+class Raw extends Bitmap {
+	/**
+	 * {@inheritDoc}
+	 */
+	public function getMimeType() {
+		return '/image\/x-dcraw/';
+	}
+}
```
Comment (nickvergessen) [inline lib/private/preview/raw.php]: same here

**Comment by oparoz** [comment]
Excellent website with samples to test with
http://www.rawsamples.ch/

**Comment by oparoz** [comment]
Looking at https://ci.owncloud.org/job/pull-request-analyser-ng-simple/9268/label=SLAVE/console , I don't think the repair tests were executed.

**Comment by LukasReschke** [comment]
Rebase required.

File: lib/private/preview/image.php [hunk 2]
```diff
@@ -13,7 +13,7 @@ class Image extends Provider {
 	 * {@inheritDoc}
 	 */
 	public function getMimeType() {
-		return '/image\/(?!tiff$)(?!svg.*).*/';
+		return '/image\/(?!x-dcraw$)(?!tiff$)(?!svg.*).*/';
```
Comment (nickvergessen) [inline lib/private/preview/image.php]: this looks wrong to me.
Seems to only work for svg as intended: https://regex101.com/r/qQ4bO5/1

Comment (oparoz) [inline lib/private/preview/image.php]: You need to test one at a time (like the Preview class does) and you should see that it works.

**Comment by MorrisJobke** [comment]
@DeepDiver1975 @nickvergessen Same as in https://github.com/owncloud/core/pull/13648#issuecomment-78261108 ?

File: lib/repair/repairmimetypes.php [hunk 3]
```diff
@@ -190,5 +214,9 @@ public function run() {
 		if ($this->fixPostscriptMimeType()) {
 			$this->emit('\OC\Repair', 'info', array('Fixed Postscript mime types'));
 		}
+
+		if ($this->introduceRawMimeType()) {
+			$this->emit('\OC\Repair', 'info', array('Fixed Postscript mime types'));
```
Comment (MorrisJobke) [inline lib/repair/repairmimetypes.php]: Copy and paste error?

**Comment by MorrisJobke** [comment]
rebase required

File: lib/repair/repairmimetypes.php (hunk 3 above)
Comment (oparoz) [inline lib/repair/repairmimetypes.php]: Good eye :)

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

**Comment by oparoz** [comment]
Failing tests are unrelated

**Comment by nickvergessen** [comment]
@owncloud-bot retest this please

**Comment by oparoz** [comment]
So why can't the bot even merge this PR?

**Comment by LukasReschke** [comment]
Bug in the CI plugin. Either retry or if that does not work create a new PR :-(

**Comment by oparoz** [comment]
;(

I just did a rebase and it failed, so I guess I'll have to create a fresh one... Thanks!

**Comment by nickvergessen** [comment]
> I just did a rebase and it failed, so I guess I'll have to create a fresh one... Thanks!

Before pushing again to a branch, make sure to wait until jenkins reported back

**Comment by oparoz** [comment]
Yep. I noticed yesterday that my default setting for push was not set to simple, so sometimes, changes were pushed to side branches as well, killing jenkins in the process if he was already looking at that branch.

So if that every happens, then it just kills the branch?

**Comment by MorrisJobke** [comment]
@oparoz You can simply close this PR and open a new one without deleting the branch - everything right on github. This should solve the Jenkins plugin problem ;)

**Comment by oparoz** [comment]
New PR #14878
Let's hope this works

**Comment by MorrisJobke** [comment]
@oparoz Looks good so far: https://ci.owncloud.org/job/pull-request-analyser-ng-simple/10443/label=SLAVE/console :)

**Comment by oparoz** [comment]
:relieved:

**Pull request closed without merging**